Add SET_DIGIPOT command to mcp4018 implementation

jake-b opened the pull request:
Added a `SET_DIGIPOT` command to the mcp4018 implementation.

Previously the mcp4018 was read only, and set at the time of configuration.  This allows you to change the value during a print, which is needed for some older printers that need to lower the stepper current during preheating. (smaller power supplies needed all the power to heat the bed, so the steppers are lowered to just enough to hold position while the bed comes up to temp, and then raised for printing.)

By way of testing, I printed an entire Voron 2.4 on a heavily modified Makerbot Replicator 2X with a Mightyboard Rev G that uses the mcp4018 to control stepper current.

Documentation updates are included, along with instructions on how to implement a `G130` macro, for compatibility with the start g-code that is commonly used with vintage printers that use mcp4018 to adjust their vRef voltages.

Thanks for your consideration.

KevinOConnor commented:
Thanks.  In general it looks fine to me.  As a minor thing, I'd ask that we not add complex macros to G-Codes.md .  If you feel an example macro is needed then maybe add it to config/sample-macros.cfg (or similar).  Alternatively, if there is a particular Klipper printer config that needs this feature then maybe add the macro to that printer config.

Also, please use the existing `gcmd.getfloat()` range checks instead of implementing them manually (eg, `gcmd.get_float('WIPER', minval=0., maxval=self.scale)`).

-Kevin

jake-b commented:
Thanks Kevin.  I've moved the sample macro out of `G-Codes.md` and into `sample-macros.cfg`.  A full printer config is probably outside the scope of this pull request for now.  Thanks for showing me the `get_float` range check parameters, also implemented.

KevinOConnor commented:
Thanks.

-Kevin
